frontend/packages/dev-console/src/components/taskruns/TaskRunLog.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { PodLogs } from '@console/internal/components/pod-logs';
+import { StatusBox } from '@console/internal/components/utils/status-box';
+import { PodModel } from '@console/internal/models';
+
+export type TaskRunLogProps = {
+  obj: K8sResourceKind;
+};
+
+const TaskRunLog: React.FC<TaskRunLogProps> = ({ obj }) => {
+  const [pod, podLoaded, podError] = useK8sGet<K8sResourceKind>(
+    PodModel,
+    obj.status.podName,
+    obj.metadata.namespace,
+  );
+  if (podLoaded) {
+    return <PodLogs obj={pod} />;
+  }
+  return <StatusBox loaded={podLoaded} loadError={podError} label="TaskRun Log" />;
+};
+
+// to safeguard against cases where Pod is yet to be assigned or is removed
+export default (props: TaskRunLogProps) =>
+  props.obj.status?.podName ? (
+    <TaskRunLog {...props} />
+  ) : (
+    <StatusBox loadError="Pod not found" label="TaskRun Log" />
```

Comment:
Why can't we have this check in the above function and early exit if podName is not available :thinking:

Reply:
It works, but it fails lint check when there is an `if` block before the `useK8sGet` Hook

Reply:
Best to name this wrapper component and export that as default.

Reply:
named as wrapper